### PR TITLE
chore(typescript): add typescript to toolbelt

### PR DESCRIPTION
**Pull request by SteveBunlon**
### General

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Test manually the implemented changes
- [x] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [x] Consider the security impact of the changes made

### CODE REVIEW

**Comment by SteveBunlon** [comment]
This PR also needs to update to github actions. They are incoming.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
           fetch-depth: 0
```
Comment (SteveBunlon) [inline .github/workflows/build.yml:17]: @arnaud-moncel @jeffladiray I'm pretty sure you could have useful insights / comments on what I've changed. Otherwise, if you could point me to someone helpful, that would be awesome ❤️

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
...
       - uses: actions/setup-node@v2
         with:
-          node-version: 14.17.6
+          node-version: 16.14.0
```
Comment (jeffladiray) [inline .github/workflows/build.yml:20]: Was this change needed?

No issue for me to keep it in this PR, but I'm wondering since we're migrating to node 18 if it shouldn't be included here

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -69,11 +69,35 @@ jobs:
       - name: Test postpack
         run: yarn postpack
 
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    needs: [test]
```
Comment (jeffladiray) [inline .github/workflows/build.yml:75]: Shouldn't we build and test in parallel, then trigger the release?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -92,6 +116,10 @@ jobs:
         run: |
           VERSION=`node -p "require('./package.json').devDependencies['semantic-release']"`
           echo "::set-output name=version::$VERSION"
+      - name: Download build
```
Comment (jeffladiray) [inline .github/workflows/build.yml:119]: If we only need the artifact to publish, I think that `action/checkout@v2` can be remove ?

File: package.json [hunk 5]
```diff
@@ -77,7 +79,9 @@
     "nock": "13.0.4",
     "rimraf": "3.0.2",
     "semantic-release": "19.0.3",
-    "semantic-release-slack-bot": "3.5.2"
+    "semantic-release-slack-bot": "3.5.2",
+    "ts-jest": "^27.1.2",
```
Comment (jeffladiray) [inline package.json:83]: Not sure if that's relevent here, but on the server we usually use `swc/jest` instead of `ts-jest`

File: .github/workflows/build.yml (hunk 3 above)
Comment (SteveBunlon) [reply .github/workflows/build.yml:75]: You are totally right

Comment (SteveBunlon) [reply .github/workflows/build.yml:75]: done, now testing and building needs lint (for energy saving purposes) and release needs build and test (they are run in parralele)

File: .github/workflows/build.yml (hunk 2 above)
Comment (SteveBunlon) [reply .github/workflows/build.yml:20]: Actually you are right, I'll remove this from this PR, and let's focus on node versions later on, this PR is primarily about typescript 👌

File: .github/workflows/build.yml (hunk 4 above)
Comment (SteveBunlon) [reply .github/workflows/build.yml:119]: To be honest, I don't know. How is the release performed ? Will the action merge the whitelisted content from the github repo + the cached dist folder ? 

This is something I can test actually, can you remind me the steps to attempt a release on beta ?

Comment (SteveBunlon) [reply .github/workflows/build.yml:120]: Done

File: package.json (hunk 5 above)
Comment (SteveBunlon) [reply package.json]: Done thanks to @Thenkei
